motto/agents/retrieval_agent.py

```diff
@@ -0,0 +1,104 @@
+import shutil
+
+import chromadb
+import numpy as np
+import os
+import pathlib
+import concurrent.futures
+
+from .agent import Agent
+from .agent import Response
+from .data_processors import OpenAIEmbeddings, DocProcessor
+from hyperon import *
+from llm_gate import atom2msg
+
+
+class RetrievalAgent(Agent):
+    max_length = 2270
+
+    def __init__(self, data_source, chunk_token_size, docs_count, data_dir):
+
+        data_source = atom2msg(data_source).strip()
+        if not (os.path.isfile(data_source) or os.path.isdir(data_source)):
+            raise AttributeError("data_source should be file or folder")
+
+        if not os.path.exists(data_source):
+            return
+
+        self.docs_count = int(atom2msg(docs_count).strip())
+
+        data_dir = atom2msg(data_dir).strip()
```

Comment:
Why is it needed with `unwrap=True`?

Reply:
> Why is it needed with `unwrap=True`?

Yes, there is no need in atom2msg here. I have some problems with functions in sent-assistant, and add this changes, but the problem was not in retrieval agent